lib/app-utils.js

```diff
@@ -88,24 +109,49 @@ async function fetchSupportedAppPlatforms(app) {
  * @param {PlatformOpts} expectedPlatform
  * @throws {Error} If bundle architecture does not match the expected device architecture.
  */
-async function verifyApplicationPlatform(app, expectedPlatform) {
+export async function verifyApplicationPlatform(app, expectedPlatform) {
   log.debug('Verifying application platform');
 
-  const supportedPlatforms = await fetchSupportedAppPlatforms.bind(this)(app);
+  const supportedPlatforms = await fetchSupportedAppPlatforms(app);
   log.debug(`CFBundleSupportedPlatforms: ${JSON.stringify(supportedPlatforms)}`);
 
   const {isSimulator, isTvOS} = expectedPlatform;
   const prefix = isTvOS ? 'AppleTV' : 'iPhone';
   const suffix = isSimulator ? 'Simulator' : 'OS';
   const dstPlatform = `${prefix}${suffix}`;
+  const appFileName = path.basename(app);
   if (!supportedPlatforms.includes(dstPlatform)) {
     throw new Error(
       `${
         isSimulator ? 'Simulator' : 'Real device'
-      } architecture is unsupported by the '${app}' application. ` +
+      } architecture is not supported by the '${appFileName}' application. ` +
         `Make sure the correct deployment target has been selected for its compilation in Xcode.`,
     );
   }
+  if (isSimulator) {
+    const executablePath = path.resolve(app, await extractExecutableName(app));
+    const [resFile, resUname] = await B.all([
+      exec('file', [executablePath]),
+      exec('uname', ['-m']),
+    ]);
+    const bundleExecutableInfo = _.trim(resFile.stdout);
+    log.debug(bundleExecutableInfo);
+    const arch = _.trim(resUname.stdout);
+    const isAppleSilicon = os.cpus[0].model.includes('Apple');
```

Comment:
Do we need this as well? In case we get the uname, might `arm64` or `x86_64` (intel mac) be sufficient?

Reply:
`arch !== 'x86_64'` for `isAppleSilicon` or `os.cpus()[0]` instead of `os.cpus[0]` worked.

Reply:
Hm.. i see. it could probably occur when a user uses node for v86_64 and runs Appium with it on Apple silicon